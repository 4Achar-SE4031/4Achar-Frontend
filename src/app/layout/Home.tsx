import React from 'react'
import FiveEvents from '../../features/events/FiveEvents'
import { FooterComponent } from './Footer'

const Home = () => {
  return (
    <div>
      <FiveEvents />
      <FooterComponent />
    </div>
  )
}

export default Home
