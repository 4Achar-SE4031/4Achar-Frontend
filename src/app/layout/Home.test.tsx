// Home.test.tsx

import React from 'react';
import { describe, it, expect, vi, beforeEach } from 'vitest';
import { render, screen } from '@testing-library/react';
import Home from './Home';
import { MemoryRouter } from 'react-router-dom';

// Mock subcomponents to isolate Home component tests
vi.mock('../../features/Navbar/navbar.tsx', () => ({
  default: () => <div data-testid="navbar">Navbar Component</div>,
}));

vi.mock('./HeroSection', () => ({
  default: () => <div data-testid="hero-section">HeroSection Component</div>,
}));

vi.mock('../../features/events/FiveEvents', () => ({
  default: () => <div data-testid="five-events">FiveEvents Component</div>,
}));

vi.mock('./Footer', () => ({
  default: () => <div data-testid="footer">FooterComponent</div>,
}));

describe('Home Component', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('renders Navbar component', () => {
    render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    const navbar = screen.getByTestId('navbar');
    expect(navbar).toBeInTheDocument();
    expect(navbar).toHaveTextContent('Navbar Component');
  });

  it('renders HeroSection component', () => {
    render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    const heroSection = screen.getByTestId('hero-section');
    expect(heroSection).toBeInTheDocument();
    expect(heroSection).toHaveTextContent('HeroSection Component');
  });

  it('renders FiveEvents component', () => {
    render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    const fiveEvents = screen.getByTestId('five-events');
    expect(fiveEvents).toBeInTheDocument();
    expect(fiveEvents).toHaveTextContent('FiveEvents Component');
  });

  it('renders FooterComponent', () => {
    render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    const footer = screen.getByTestId('footer');
    expect(footer).toBeInTheDocument();
    expect(footer).toHaveTextContent('FooterComponent');
  });



  it('matches the snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  // Additional Test: Accessibility Check

});
