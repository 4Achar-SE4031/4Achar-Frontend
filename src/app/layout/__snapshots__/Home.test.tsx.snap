// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      data-testid="navbar"
    >
      Navbar Component
    </div>
    <div
      class="home-page"
    >
      <div
        id="mainScrollContainer"
      >
        <div
          data-testid="hero-section"
        >
          HeroSection Component
        </div>
        <div
          data-testid="five-events"
        >
          FiveEvents Component
        </div>
        <div
          data-testid="footer"
        >
          FooterComponent
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
